packages/sanity/src/core/studio/colorScheme.tsx

```diff
@@ -155,22 +155,24 @@ export function ColorSchemeCustomProvider({
 }
 
 /** @alpha */
-export function useColorSchemeSetValue() {
+export function useColorSchemeSetValue():
+  | false
+  | ((nextScheme: StudioThemeColorSchemeKey) => void) {
   const setValue = useContext(ColorSchemeSetValueContext)
   if (setValue === null) throw new Error('Could not find `ColorSchemeSetValueContext` context')
   return setValue
 }
 
 /** @internal */
-export function _useColorSchemeInternalValue(): StudioThemeColorSchemeKey {
```

Comment:
do you think this hook needs to be exported? would refactoring it to not be exported be better? it seems like it was prefaced with an _ to prevent external usage but not exposing it does the same job

Reply:
I'd be happy to do that in a quick follow-up! It's only needed internally so it doesn't need to be available on `import 'sanity'`

Reply:
Btw the underscore caused the React Compiler to not see it as a hook, causing hook re-ordering errors, which is why it was changed here 🙌

Reply:
Good to know 💡